infrastructure/index.ts

```diff
@@ -50,43 +52,28 @@ const contentBucket = new aws.s3.Bucket(
     },
 );
 
-// contentBucket needs to have the "public-read" ACL so its contents can be ready by CloudFront and
-// served. But we deny the s3:ListBucket permission to prevent unintended disclosure of the bucket's
-// contents.
-const denyListPolicyState: aws.s3.BucketPolicyArgs = {
-    bucket: contentBucket.bucket,
-    policy: contentBucket.arn.apply((arn: string) => JSON.stringify({
+// The website bucket needs to have the "public-read" ACL so its contents can be read by
+// CloudFront and served. But we deny the s3:ListBucket permission to anyone but account
+// users to prevent unintended disclosure of the bucket's contents.
+const policy = new aws.s3.BucketPolicy("website-bucket-policy", {
+    bucket: websiteBucket.bucket,
+    policy: websiteBucket.arn.apply(arn => JSON.stringify({
         Version: "2008-10-17",
         Statement: [
             {
                 Effect: "Deny",
                 Principal: "*",
                 Action: "s3:ListBucket",
                 Resource: arn,
+                Condition: {
+                    StringNotEquals: {
+                        "aws:PrincipalAccount": config.principalAccount,
```

Comment:
Why don't we pull the account ID at runtime? So that it always matches the AWS account used, and "owns" the bucket? e.g.
https://www.pulumi.com/docs/reference/pkg/nodejs/pulumi/aws/#getCallerIdentity

Hooking it up via config just means that it's something that we could get wrong. (Configure to use account X, but actually deploy to account Y.) By pulling this at runtime, it will always "just work". Right?

Reply:
That sounds great! How do o do that? 😉